src/com/liberty_amls/PlatformHandler.java

```diff
@@ -335,4 +340,42 @@ public void stop() {
         handleRunning = false;
         disableLight();
     }
+
+    /**
+     * This method calculates current platform's velocity in km/h
+     */
+    private void calculateSpeed(){
+        double loopTime = System.currentTimeMillis() - platformLastPacketTime;
+
+        double distance = Math.sin((platformContainer.trueGPSLat.get(platformContainer.trueGPSLat.size() - 1) -

```

Comment:
When I talked about reducing the number of local variables, I meant only those places where they are not needed. It is too irrational to count the size of the array every time. Please declare a variable for this